ledger/internal/eval.go

```diff
@@ -251,10 +254,13 @@ func (x *roundCowBase) lookupAssetParams(addr basics.Address, aidx basics.AssetI
 		return ledgercore.AssetParamsDelta{}, false, fmt.Errorf("lookupAssetParams couldn't find addr %s aidx %d in cache: %w", addr.String(), aidx, ErrNotInCowCache)
 	}
 
-	resourceData, err := x.l.LookupAsset(x.rnd, addr, aidx)
+	resourceData, rnd, err := x.l.LookupAsset(addr, aidx)
 	if err != nil {
 		return ledgercore.AssetParamsDelta{}, false, err
 	}
+	if rnd != x.rnd {
+		return ledgercore.AssetParamsDelta{}, false, ledgercore.ErrNonSequentialBlockEval{EvaluatorRound: x.rnd, LatestRound: rnd}
```

Comment:
Please add a test this new error is correctly forwarded to the evaluator's caller and handled there?